### ISSUE TITLE
Stuck in "opening stream" window

### ISSUE THREAD

**Issue by albertfc** [scakemyer/plugin.video.quasar#569]
Quasar always get stuck in "opening stream" window when trying to play something. After almost 5 minutes it fails and here it is the log:

```
[...] 
22:34:44 T:140123316623104  NOTICE: [script.module.magnetic] Providers search returned: 20 results in 8.8 seconds
22:34:44 T:140123316623104  NOTICE: [script.module.magnetic] Filtering returned: 20 results
[...]
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar] Traceback (most recent call last):
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]   File "/home/xbmc/.kodi/addons/plugin.video.quasar/resources/site-packages/quasar/navigation.py", line 112, in run
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]     data = _json(url)
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]   File "/home/xbmc/.kodi/addons/plugin.video.quasar/resources/site-packages/quasar/navigation.py", line 48, in _json
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]     with closing(urllib2.urlopen(url)) as response:
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]   File "/usr/lib/python2.7/urllib2.py", line 127, in urlopen
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]     return _opener.open(url, data, timeout)
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]   File "/usr/lib/python2.7/urllib2.py", line 404, in open
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]     response = self._open(req, data)
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]   File "/usr/lib/python2.7/urllib2.py", line 422, in _open
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]     '_open', req)
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]   File "/usr/lib/python2.7/urllib2.py", line 382, in _call_chain
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]     result = func(*args)
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]   File "/usr/lib/python2.7/urllib2.py", line 1214, in http_open
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]     return self.do_open(httplib.HTTPConnection, req)
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]   File "/usr/lib/python2.7/urllib2.py", line 1187, in do_open
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]     r = h.getresponse(buffering=True)
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]   File "/usr/lib/python2.7/httplib.py", line 1089, in getresponse
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]     response.begin()
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]   File "/usr/lib/python2.7/httplib.py", line 444, in begin
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]     version, status, reason = self._read_status()
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]   File "/usr/lib/python2.7/httplib.py", line 400, in _read_status
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]     line = self.fp.readline(_MAXLINE + 1)
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]   File "/usr/lib/python2.7/socket.py", line 476, in readline
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]     data = self._sock.recv(self._rbufsize)
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar] timeout: timed out
22:39:34 T:140122360112896   ERROR: [plugin.video.quasar]
```

**Comment by mancuniancol** [comment]
Try with one provider at the time and reduce the number of results in each provider from. 10 to 5

**Comment by scakemyer** [comment]
Also getting this, but only with episodes... Using the "magnetizer" gets around the issue, so I'm looking into Quasar's resolver for trackers, which is around where is seems to get stuck, but I have the impression that Magnetic is doing something funky with the results it's sending back to Quasar...

**Comment by mancuniancol** [comment]
@scakemyer, you are right.  Some providers needs to do two or more request to get the torrent or magnet information.  Initially, those providers were too slow more than 8s.  Then, they were not practical.

As solution, I've create a service inside the Magnetic to read the information an provide a torrent which reduce the response time below 3s.  

The problem comes that Magnetic cannot give a magnet in the second call. It needs to be torrent file.  However, some places uses itorrent.org as torrent storage or there is not more information than the hash (Magnetic creates the torrent from that site too).  For some reason, sometimes, itorrent.org doesn't provide the torrent, but a webpage.  It could be publicity or a type of blocking.  I used to use torcache, but it is gone for good.

The difference with Magnetizer is that Magnetic can send magnet information.  Then, the problem with the itorrent.org is not present.

Conclusion, not to use the providers which have the open subpage enable with Quasar.

**Comment by albertfc** [comment]
Didn't know the "magnetizer" workarround! Thanks anyway!

**Comment by scakemyer** [comment]
Fixed for 0.9.37 so we won't depend on Magnetic or trackers playing nice :)
